packages/typescript/internal/ts_project.bzl

```diff
@@ -98,6 +122,19 @@ def _calculate_root_dir(ctx):
     )
 
 def _ts_project_impl(ctx):
+    srcs = [src.path for src in ctx.files.srcs]
+    typings_out_dir = ctx.attr.declaration_dir or ctx.attr.out_dir
+
+    # Recalculate outputs inside the rule implementation.
+    # The outs are first calculated in the macro in order to try to predetermine outputs so they can be declared as
+    # outputs on the rule. This provides the benefit of being able to reference an output file with a label.
+    # However, it is not possible to evaluate files in outputs other rules such as filegroup, therefore the outs a
```

Comment:
outputs other rules -> outputs of other rules
outs a -> outs are

Reply:
Will fix the comment

Reply:
?

Reply:
Fixed now